src/praisonai-agents/praisonaiagents/snapshot/snapshot.py

```
@@ -414,26 +430,37 @@ def restore(self, commit_hash: str, files: Optional[List[str]] = None) -> bool:
+                    set(ls.stdout.strip().split("\n"))
+                    if ls.stdout.strip()
+                    else set()
+                )
                 self._run_git("checkout", commit_hash, "--", ".")
                 
-                # Copy all files back to project
-                for root, dirs, files_list in os.walk(self.shadow_path):
-                    # Skip .git directory
-                    if ".git" in root:
+                # Remove project files not present in the target commit.
+                for root, dirs, files_list in os.walk(self.project_path):
```

Ruff (0.16.0)

[warning] Loop control variable `dirs` not used within loop body

Rename unused `dirs` to `_dirs`

(B007)